module/validation/receipt_validator.go

```diff
@@ -88,11 +88,11 @@ func (v *receiptValidator) verifyChunksFormat(result *flow.ExecutionResult) erro
 	requiredChunks := 1 // system chunk: must exist for block's ExecutionResult, even if block payload itself is empty
 
 	index, err := v.index.ByBlockID(result.BlockID)
-	if err != nil {
-		if !errors.Is(err, storage.ErrNotFound) {
-			return err
-		}
-		// reaching this line means the block is empty, i.e. it has no payload => we expect only the system chunk
+	if errors.Is(err, storage.ErrNotFound) {
+		// the block is empty, i.e. it has no payload => we expect only the system chunk
```

Comment:
I might happen to catch a bug here, that we should return an error here instead. because even if the block has no payload, we should still [have an entry stored](https://github.com/onflow/flow-go/blob/master/storage/badger/payloads.go#L61) in the index db